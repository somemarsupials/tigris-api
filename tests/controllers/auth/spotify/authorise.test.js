const route = require('../../../../controllers/auth/spotify/authorise');

describe('auth/spotify/authorise', () => {
  let req, res;

  beforeEach(() => {
    req = {
      origin: 'site.com',
      provider: {
        clientId: 0,
        authoriseURI: 'auth.com'
      },
      session: {
        id: 1
      },
    };
    res = { redirect: jest.fn() };
    route(req, res, null);
  });

  it('redirects correctly', () => {
    expect(res.redirect.mock.calls).toMatchSnapshot();
  });

  it('calls redirect once', () => {
    expect(res.redirect).toHaveBeenCalledTimes(1);
  });
});
