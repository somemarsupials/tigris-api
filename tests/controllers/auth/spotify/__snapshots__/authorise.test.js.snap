// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth/spotify/authorise redirects correctly 1`] = `
Array [
  Array [
    "auth.com?response_type=code&scopes=user-read-private%20user-read-email&client_id=0&redirect_uri=&state=1",
  ],
]
`;
